lib/nats.js

```diff
@@ -1007,7 +1050,7 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   if (opt_callback !== undefined) {
     this.flush(opt_callback);
   } else if (this.closed) {
-    throw(new Error(CONN_CLOSED));
+    throw(new NatsError(CONN_CLOSED_MSG, CONN_CLOSED));
```

Comment:
This is what I suggested on #104 

Seems like a good approach too, instanceof can be used to handle the errors.